feat: docs directory added (#130)
codex-team/notes.api: docs/inheritance-policy.md

> #Inheritance policy

Reviewer: ```suggestion
# Inheritance policy
```

> For each parameter calculation calculations are made in this order:

Reviewer: ```suggestion
For each parameter calculations are made in this order:
```

> - if parameter in note-settings has is's own value (excluding None, Undefined, etc.), then this field does not inherite this parameter from parent.

Reviewer: ```suggestion

- if parameter in note-settings has is's own value (excluding None, Undefined, etc.), then this field does not inherite this parameter from parent.
```

> - if parameter in note-settings has is's own value (excluding None, Undefined, etc.), then this field does not inherite this parameter from parent.

Reviewer: ```suggestion
- if parameter in `note-settings` has is's own value, then this field does not inherits this parameter from parent.
```

> - if the parameter is in default position or has a value Undefined, None, etc., then its value is calculated as the value of the same parameter in the parent_settings of the parent note.

Reviewer: ```suggestion
- if the parameter has the default value (Undefined, None, etc.), then its value is calculated as the value of the same parameter in the `note_settings` of the parent note.
```

> - if parent note has this parameter, set to default, then algorithm checks this value in parent note of the parent note.

Reviewer 2: The 3rd point is not quite clear. Why do we need to go to the parent note of the parent note? Judging by point 1 and 2, the `default` field can only be if all previous parent notes had this `default` field - so why should we check them? Can you explain, please?

Author: this is kind of dynamic field, if parent has `undefined` value in some parameter, that means, that we should check in older parent (`undefined` works as ref to parent's value)
